lib/mutant/ast/regexp/transformer/direct.rb

```diff
@@ -0,0 +1,99 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        class Direct < self
+          %i[
+            regexp_one_or_more_escape
+            regexp_group_open_escape
+            regexp_group_close_escape
+            regexp_interval_open_escape
+            regexp_interval_close_escape
+            regexp_newline_escape
+            regexp_zero_or_more_escape
+            regexp_carriage_escape
+            regexp_dot_escape
+            regexp_set_open_escape
+            regexp_set_close_escape
+            regexp_eol_escape
+            regexp_bos_anchor
+            regexp_match_start_anchor
+            regexp_word_boundary_anchor
+            regexp_eos_ob_eol_anchor
+            regexp_eos_anchor
+            regexp_bol_anchor
+            regexp_eol_anchor
+            regexp_nonword_boundary_anchor
+            regexp_alpha_property
+            regexp_script_arabic_property
+            regexp_script_hangul_property
+            regexp_script_han_property
+            regexp_script_hiragana_property
+            regexp_script_katakana_property
+            regexp_letter_any_property
+            regexp_digit_type
+            regexp_space_type
+            regexp_word_type
+            regexp_nondigit_type
+            regexp_nonspace_type
+            regexp_nonword_type
+            regexp_dot_meta
+          ].map(&method(:register))
+
+          class ExpressionToAST < Transformer::ExpressionToAST
+            def call
+              quantify(ast)
+            end
+          end
+
+          class ASTToExpression < Transformer::ASTToExpression
+            include LookupTable
+
+            # rubocop:disable LineLength
+            TABLE = IceNine.deep_freeze(
+              regexp_one_or_more_escape:       [::Regexp::Token.new(:escape,   :one_or_more,      '\+'),           ::Regexp::Expression::EscapeSequence::Literal],
```

Comment:
I dunno if its worth it, but we may be able to remove the repeated `::Regexp::Token.new` calls via placing the arguments in the value array and later mapping over the values to instantiate the token object.

This reduces literall call side duplication.

```ruby
[
  [:regexp_one_or_more_escape, [:escape, :one_or_more, '\+'], ::Regexp::Expression::EscapeSequence::Literal],
  ...
].map { |type, arguments, klass| [type, [::Regexp::Token.new(*arguments), klass] }.to_h
```

Will also help to reduce flay / flog score increases.